File: app/src/main/java/lemond/annoying/gamerscompanion/fragment_news/viewmodel/PublishedTimeFromNow.java
package lemond.annoying.gamerscompanion.fragment_news.viewmodel;


public class PublishedTimeFromNow {

    public enum TimeUnit {
        DAY,
        HOUR,
        MINUTE,
        SECOND,
        MOMENT
    }

    public TimeUnit timeUnit;
    public int timeDuration = 0;

    PublishedTimeFromNow(long publishedTimeMillis) {
        long now = System.currentTimeMillis();
        long diff = now - publishedTimeMillis;

        long secondsInMilli = 1000;
        long minutesInMilli = secondsInMilli * 60;
        long hoursInMilli = minutesInMilli * 60;
        long daysInMilli = hoursInMilli * 24;

        long numberDuration;

        if ((numberDuration = (int) diff / daysInMilli) > 0) {
            timeUnit = TimeUnit.DAY;
            timeDuration = (int) numberDuration;
        } else if ((numberDuration = diff / hoursInMilli) > 0) {
            timeUnit = TimeUnit.HOUR;
            timeDuration = (int) numberDuration;
        } else if ((numberDuration = diff / minutesInMilli) > 0) {
            timeUnit = TimeUnit.MINUTE;
            timeDuration = (int) numberDuration;
        } else if ((numberDuration = diff / secondsInMilli) > 0) {
            timeUnit = TimeUnit.SECOND;
            timeDuration = (int) numberDuration;
        } else {
            timeUnit = TimeUnit.MOMENT;
        }
    }
}

File: app/src/main/java/lemond/annoying/gamerscompanion/main_activity/fragment_news/viewmodel/PublishedTimeFromNow.kt
package lemond.annoying.gamerscompanion.main_activity.fragment_news.viewmodel


class PublishedTimeFromNow internal constructor(publishedTimeMillis: Long) {

    var timeUnit: TimeUnit
    var timeDuration = 0

    enum class TimeUnit {
        DAY,
        HOUR,
        MINUTE,
        SECOND,
        MOMENT
    }

    init {
        val now = System.currentTimeMillis()
        val diff = now - publishedTimeMillis

        val secondsInMilli: Long = 1000
        val minutesInMilli = secondsInMilli * 60
        val hoursInMilli = minutesInMilli * 60
        val daysInMilli = hoursInMilli * 24

        when {
            (diff.toInt() / daysInMilli) > 0 -> {
                timeUnit = TimeUnit.DAY
                val numberDuration: Long = (diff.toInt() / daysInMilli)
                timeDuration = numberDuration.toInt()

            }
            (diff / hoursInMilli) > 0 -> {
                timeUnit = TimeUnit.HOUR
                val numberDuration: Long = (diff.toInt() / hoursInMilli)
                timeDuration = numberDuration.toInt()

            }
            (diff / minutesInMilli) > 0 -> {
                timeUnit = TimeUnit.MINUTE
                val numberDuration: Long = (diff.toInt() / minutesInMilli)
                timeDuration = numberDuration.toInt()

            }
            (diff / secondsInMilli) > 0 -> {
                timeUnit = TimeUnit.SECOND
                val numberDuration: Long = (diff.toInt() / secondsInMilli)
                timeDuration = numberDuration.toInt()

            }
            else -> timeUnit = TimeUnit.MOMENT
        }
    }
}
